Indexing/ClauseCodeTree.cpp

```diff
@@ -646,17 +646,17 @@ inline bool ClauseCodeTree::ClauseMatcher::canEnterLiteral(CodeOp* op)
   }
 
   if(ils->isVarEqLit) {
-    unsigned idxVarSort = ils->varEqLitSort;
+    TermList idxVarSort = ils->varEqLitSort;
     size_t matchIndex=ils->matchCnt;
     while(matchIndex!=0) {
       matchIndex--;
       MatchInfo* mi=ils->getMatch(matchIndex);
       unsigned liIntex = mi->liIndex;
       Literal* lit = (*query)[lInfos[liIntex].litIndex];
       ASS(lit->isEquality());
-      unsigned argSort = SortHelper::getEqualityArgumentSort(lit);
-      if(idxVarSort!=argSort) {
-	ils->deleteMatch(matchIndex); //decreases ils->matchCnt
+      TermList argSort = SortHelper::getEqualityArgumentSort(lit); 
+      if(idxVarSort!=argSort) {//TODO check that this is what we want. Perhaps require unification
```

Comment:
I forgot, can we resolved this with current knowledge?

Reply:
Unfortunately not. I don't understand CodeTrees, so I am not sure what this function does, or what ramifications changing away from syntactic equality check may have. The good news is that the only place we use ClauseCodeTrees is in demodulation, so having a too strict test here will at worst result in some lost simplifications.

Reply:
OK, then let's leave a TODO